src/Tgstation.Server.Host/Components/Chat/Providers/IrcProvider.cs

```diff
@@ -693,5 +692,44 @@ await Task.WhenAny(
 					listenTask ?? Task.CompletedTask),
 				AsyncDelayer.Delay(TimeSpan.FromSeconds(5), cancellationToken));
 		}
+
+		/// <summary>
+		/// Creates a new instance of the IRC client.
+		/// Reusing the same client after a disconnection seems to cause issues.
+		/// </summary>
+		/// <returns>The <see cref="IrcFeatures"/> client to use.</returns>
+		private IrcFeatures InstantiateClient()
+		{
+			IrcFeatures newClient = new IrcFeatures
+			{
+				SupportNonRfc = true,
+				CtcpUserInfo = "You are going to play. And I am going to watch. And everything will be just fine...",
+				AutoRejoin = true,
+				AutoRejoinOnKick = true,
+				AutoRelogin = false,
+				AutoRetry = false,
+				AutoReconnect = false,
+				ActiveChannelSyncing = true,
+				AutoNickHandling = true,
+				CtcpVersion = versionString,
+				UseSsl = ssl,
+				EnableUTF8Recode = true,
+			};
+			if (ssl)
+				newClient.ValidateServerCertificate = true; // dunno if it defaults to that or what
+
+			newClient.OnChannelMessage += Client_OnChannelMessage;
+			newClient.OnQueryMessage += Client_OnQueryMessage;
+
+			newClient.OnReadLine += (sender, e) => Logger.LogTrace("READ: {line}", e.Line);
+			newClient.OnWriteLine += (sender, e) => Logger.LogTrace("WRITE: {line}", e.Line);
```

Comment:
Can you comment these two out again? They are really spammy for the logs in active channels.
```suggestion
			/*newClient.OnReadLine += (sender, e) => Logger.LogTrace("READ: {line}", e.Line);
			newClient.OnWriteLine += (sender, e) => Logger.LogTrace("WRITE: {line}", e.Line);*/
```

Reply:
Is there any way we could make a config boolean to enable this? 
from experience, it can be extremely helpful to diagnose issues.
having to compile from source for this could be fairly cumbersome esp. if trouble happens in prod.

Reply:
Commented in latest commit

Reply:
Yeah, you could modify the `GeneralConfiguration` to put this behind a flag.

Reply:
Actually, this is more appropriate for the `FileLoggingConfiguration`